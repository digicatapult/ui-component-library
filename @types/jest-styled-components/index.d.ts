declare module 'jest-styled-components/serializer'
