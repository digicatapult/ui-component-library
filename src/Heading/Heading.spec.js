import React from 'react'
import styled from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { H1, H2, H3, H4, H5, H6, H6, HX } from './index.js'

describe('Heading', () => {
  test('H1', () => {
    const tree = renderer.create(<H1 id="id1">Hello1</H1>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H2', () => {
    const tree = renderer.create(<H2 id="id2">Hello2</H2>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H3', () => {
    const tree = renderer.create(<H3 id="id3">Hello3</H3>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H4', () => {
    const tree = renderer.create(<H4 id="id4">Hello4</H4>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H5', () => {
    const tree = renderer.create(<H5 id="id5">Hello5</H5>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H6', () => {
    const tree = renderer.create(<H6 id="id6">Hello6</H6>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  for (let i = 1; i <= 6; i++) {
    test(`HX - h${i}`, () => {
      const tree = renderer
        .create(
          <HX headingLevel={i} id={`idx${i}`}>
            Hello-x{i}
          </HX>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    test(`Restyled HX - h${i}`, () => {
      const tree = renderer
        .create(
          <RestyledHX headingLevel={i} id={`idx${i}`}>
            Hello-x{i}
          </RestyledHX>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})

const RestyledHX = styled(HX)`
  font-size: 2em;
`
