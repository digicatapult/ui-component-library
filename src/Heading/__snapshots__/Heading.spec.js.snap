// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading H1 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h1
  className="c0"
  id="id1"
>
  Hello1
</h1>
`;

exports[`Heading H2 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h2
  className="c0"
  id="id2"
>
  Hello2
</h2>
`;

exports[`Heading H3 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h3
  className="c0"
  id="id3"
>
  Hello3
</h3>
`;

exports[`Heading H4 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h4
  className="c0"
  id="id4"
>
  Hello4
</h4>
`;

exports[`Heading H5 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h5
  className="c0"
  id="id5"
>
  Hello5
</h5>
`;

exports[`Heading H6 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h6
  className="c0"
  id="id6"
>
  Hello6
</h6>
`;

exports[`Heading HX - h1 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h1
  className="c0"
  id="idx1"
>
  Hello-x
  1
</h1>
`;

exports[`Heading HX - h2 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h2
  className="c0"
  id="idx2"
>
  Hello-x
  2
</h2>
`;

exports[`Heading HX - h3 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h3
  className="c0"
  id="idx3"
>
  Hello-x
  3
</h3>
`;

exports[`Heading HX - h4 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h4
  className="c0"
  id="idx4"
>
  Hello-x
  4
</h4>
`;

exports[`Heading HX - h5 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h5
  className="c0"
  id="idx5"
>
  Hello-x
  5
</h5>
`;

exports[`Heading HX - h6 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

<h6
  className="c0"
  id="idx6"
>
  Hello-x
  6
</h6>
`;

exports[`Heading Restyled HX - h1 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h1
  className="c0 c1"
  id="idx1"
>
  Hello-x
  1
</h1>
`;

exports[`Heading Restyled HX - h2 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h2
  className="c0 c1"
  id="idx2"
>
  Hello-x
  2
</h2>
`;

exports[`Heading Restyled HX - h3 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h3
  className="c0 c1"
  id="idx3"
>
  Hello-x
  3
</h3>
`;

exports[`Heading Restyled HX - h4 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h4
  className="c0 c1"
  id="idx4"
>
  Hello-x
  4
</h4>
`;

exports[`Heading Restyled HX - h5 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h5
  className="c0 c1"
  id="idx5"
>
  Hello-x
  5
</h5>
`;

exports[`Heading Restyled HX - h6 1`] = `
.c0 {
  font-size: 1em;
  padding: 0;
  margin: 0;
}

.c1 {
  font-size: 2em;
}

<h6
  className="c0 c1"
  id="idx6"
>
  Hello-x
  6
</h6>
`;
